winpr/libwinpr/smartcard/smartcard_pcsc.c

```diff
@@ -561,11 +574,20 @@ PCSC_SCARDHANDLE* PCSC_ConnectCardHandle(SCARDCONTEXT hSharedContext, SCARDCONTE
 	pContext->dwCardHandleCount++;
 
 	if (!g_CardHandles)
+	{
 		g_CardHandles = ListDictionary_New(TRUE);
+		if (!g_CardHandles)
+			goto error;
+	}
 
-	ListDictionary_Add(g_CardHandles, (void*) hCard, (void*) pCard);
+	if (!ListDictionary_Add(g_CardHandles, (void*) hCard, (void*) pCard))
+		goto error;
 
```

Comment:
Maybe move "pContext->dwCardHandleCount++;" from line 574 above directly before the return ?